provider_shopper/test/login_widget_test.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider_shopper/screens/login.dart';
+
+void main() {
+  testWidgets('Login page Widget test', (tester) async {
+    await tester.pumpWidget(MaterialApp(home: MyLogin()));
+
+    // Testing the layout and positioning of the screen widgets
```

Comment:
This test is mostly checking that the build method is creating the basic layout correctly, which is not all that interesting of a thing to test. I can't imagine this test ever failing, for instance.

What would be a more effective test would be a check that the ENTER button does what it's supposed to do (navigate to a new route). That behavior is specific to the LoginScreen, and should definitely be tested. You could convert this test to do that by cutting everything above line 30, adding a tap on the ENTER button, doing a pumpAndSettle, and then looking for the CatalogScreen.